context/AppContextProvider.tsx

```diff
@@ -43,35 +39,49 @@ interface AppContextProviderProps {
 const AppContextProvider = ({ children }: AppContextProviderProps) => {
   const [currentUser, setCurrentUser] = useState<IUser | null>(null);
 
-  const loginUser = async () => {
-    const isInitiated= await Pi.initialized
-    try {
-      if (isInitiated) {
-        const authResult = await Pi.authenticate(["payments", "username"], onIncompletePaymentFound)
-        const res = await axiosClient.post("/api/v1/users/authenticate", {authResult})  
-        toast.success(res.data?.user?.username)
-        setCurrentUser(res.data?.user)
-        localStorage.setItem("token", res.data?.token)
-      } else {
-        console.log("PI SDK failed to initialize.")
+  const registerUser = async () => {
+    const isInitiated= await Pi.initialized;
+    if (isInitiated) {  
+      try {
+        const pioneerAuth = await Pi.authenticate(['username'], onIncompletePaymentFound) // obtain pioneer access token from sandbox
+        
+        const authResult = await PiAuthentication(pioneerAuth.accessToken);
+        console.log('pi auth result', authResult.username);
+        // Extract username and uid of the pi user
+        const user = {
+          username: authResult.username,
+          uid: authResult.uid
+        }
+        const res = await axiosClient.post("/users/authenticate", {user}) 
```

Comment:
I had to change L55 to `const res = await axiosClient.post("/users/authenticate", { auth: {user} }); ` to get the login to work.